client/lib/directly/index.js

```diff
@@ -101,7 +102,9 @@ export function initialize() {
 
 		loadScript( DIRECTLY_RTM_SCRIPT_URL, function( error ) {
 			if ( error ) {
-				reject( error );
+				reject( new Error(
+					i18n.translate( 'Failed to load script "%(src)s".', { args: { src: error.src } } )
```

Comment:
I haven't seen anything best practices in Calypso docs, but my understanding was that it's handy to reject `Error` objects because then when you `.catch( err )`, `err` will always be an `Error` whether it was manually rejected or an actual runtime error occurred.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject#Description

> The static Promise.reject function returns a Promise that is rejected. For debugging purposes and selective error catching, it is useful to make reason an instanceof Error.

If that's ok great, but I'm definitely not married to it.